osu.Game.Rulesets.Mania/UI/Components/DefaultKeyArea.cs

```diff
@@ -36,10 +36,31 @@ public DefaultKeyArea()
             RelativeSizeAxes = Axes.Both;
         }
 
+        public class ChildContainer : Container
+        {
+            private ManiaInputManager.RulesetKeyBindingContainer keyBindingContainer { get; set; }
+
+            private ManiaInputManager.RulesetKeyBindingContainer getKeyBindingContainer()
+            {
+                return keyBindingContainer ??= (ManiaInputManager.RulesetKeyBindingContainer)((ManiaInputManager)GetContainingInputManager()).KeyBindingContainer;
+            }
+
+            protected override bool OnTouchDown(TouchDownEvent e)
+            {
+                getKeyBindingContainer().TriggerPressed(((DefaultKeyArea)Parent).column.Action.Value);
+                return base.OnTouchDown(e);
+            }
+
+            protected override void OnTouchUp(TouchUpEvent e)
+            {
+                getKeyBindingContainer().TriggerReleased(((DefaultKeyArea)Parent).column.Action.Value);
+            }
+        }
+
         [BackgroundDependencyLoader]
         private void load(IScrollingInfo scrollingInfo)
         {
-            InternalChild = directionContainer = new Container
+            InternalChild = directionContainer = new ChildContainer
```

Comment:
this entire implementation is unviable. you've placed this in `DefaultKeyArea` which is a *skinned* component and used on the default skin only. meaning this won't work on legacy skins *at all*. did you test this properly?